eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Upec.hs

```diff
@@ -105,7 +100,7 @@ instance
         coreNodeQuorum <- liftSTS $ asks quorum
 
         let utxoState = lsUTxOState ls
-            pup = proposals $ utxosGovState utxoState
+            pup = proposals ppupState
```

Comment:
Is this govState `ppupState   from the  `UpecState` the same with the one coming from the environment (as previously used) ?

Reply:
Yes, if you look at the epoch rule, you'll see they are both coming from `utxoSt'`
```haskell
      ls' = ls {lsUTxOState = utxoSt', lsCertState = adjustedCertState}

  UpecState pp' ppupSt' <-
    trans @(EraRule "UPEC" era) $
      TRC (ls', UpecState pp (utxosGovState utxoSt'), ())
```

Same with the TICKF:
```haskell
          ls = esLState es
          updates = utxosGovState $ lsUTxOState ls
      UpecState pp' _ <-
        trans @(EraRule "UPEC" era) $
          TRC (ls, UpecState pp updates, ())
```

Reply:
Makes sense, thanks!